packages/xrpl/src/models/transactions/common.ts

```diff
@@ -696,3 +733,170 @@ const _DOMAIN_ID_LENGTH = 64
 export function isDomainID(domainID: unknown): domainID is string {
   return isString(domainID) && domainID.length === _DOMAIN_ID_LENGTH
 }
+
+/* eslint-disable max-lines-per-function -- Required here as structure validation is verbose. */
+/* eslint-disable max-statements -- Required here as structure validation is verbose. */
+
+/**
+ * Validates if MPTokenMetadata adheres to XLS-89d standard.
+ *
+ * @param input - Hex encoded MPTokenMetadata.
+ * @returns Validation error messages and boolean denoting if MPTokenMetadata adheres to XLS-89d standard.
+ */
+export function validateMPTokenMetadata(input: string): {
+  isValid: boolean
+  validationMessages: string[]
```

Comment:
```suggestion
  validationMessages?: string[]
```
We can infer if validity by checking if `validationMessages` is returned.